### DISCUSSION TITLE
nohup 이용 방법 공유

### DISCUSSION THREAD

**Discussion by jiwoo0212** [boostcampaitech2/object-detection-level2-cv-06#71] [ideas]
모델 돌릴 때 서버와 연결이 끊겨도 종료되지 않도록 nohup을 사용하는 방법을 정리해봤습니다. 
그냥 vscode 터미널에 아래와 같이 입력하시면 됩니다. 

- 실행
`nohup python train.py >> nohup_log.out & `

- 로그 보기
 `tail -f nohup_log.out`

- 멈추기
`ps -ef | grep python 으로 pid 확인 후 kill -9 pid번호`

- 퍼미션 요구할 시
`chmod u+x script.py`

- 여러 파일을 순차적으로 실행할 수 있도록 하는 방법도 아래 글 참고하세요!
https://aitech2.slack.com/archives/C02EXNKH8JK/p1633101486057500

아래 슬랙과 준혁님이 알려주셨습니다~
https://aitech2.slack.com/archives/C02E95B8XN2/p1633066549015900

**Comment by ppskj178** [comment]
nohup sample입니다.
```
nohup sh -c 'python /opt/ml/detection/mmdetection/tools/train.py /opt/ml/detection/mmconfig/test/[test]valid_retina_r50_fpn_pretrained.py 1> /dev/null 2>&1 && python /opt/ml/detection/mmdetection/tools/train.py /opt/ml/detection/mmconfig/test/[test]valid_faster-rcnn_pretrained.py 1> /dev/null 2>&1 && python /opt/ml/detection/mmdetection/tools/train.py /opt/ml/detection/mmconfig/test/[test]valid_cascade_rcnn_r50_fpn_pretrained.py 1> /dev/null 2>&1' &
```
